use-shopping-cart/core/stripe-middleware.js

```diff
@@ -54,7 +33,20 @@ export const handleStripe = (store) => (next) => async (action) => {
         lineItems: [{ price: productId, quantity: 1 }]
       }
       stripe.redirectToCheckout(options)
+      return
     }
+
+    throw new Error(`Invalid checkout mode '${cart.cartMode}' was chosen. Valid modes are: 'client-only'.`)
   }
+
   return next(action)
 }
+
+async function initializeStripe(publicKey) {
+  try {
+    // eslint-disable-next-line no-undef
+    return await Stripe(publicKey)
+  } catch (error) {
+    console.log('error', error)
+  }
+}
```

Comment:
I know that this is pretty much what we were doing before, but it would be preferable if we didn't just log the error and pretend it didn't happen. Perhaps we could check which Stripe error it is with `instanceof` and then log based on that and re-throw?

Reply:
Also, you don't need to use async-await if you're just returning `Stripe(publicKey)`.

```suggestion
function initializeStripe(publicKey) {
  try {
    // eslint-disable-next-line no-undef
    return Stripe(publicKey)
  } catch (error) {
    console.log('error', error)
  }
}
```